src/bin/triplydb-run-pipeline.ts

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env node
+
+import { program } from "commander";
+import colors from "colors";
+import App from "../App.js";
+import Pipeline, { createPipeline, defaultLogger } from "../Pipeline.js";
+import { Models } from "@triply/utils";
+import fs from "fs-extra";
+
+let defaultTriplyDBToken = process.env["TRIPLYDB_TOKEN"];
+let defaultTriplyDBAccount = process.env["TRIPLYDB_ACCOUNT"];
+let defaultHttpsProxy = process.env["HTTPS_PROXY"];
+let defaultHttpProxy = process.env["HTTP_PROXY"];
+const timeLog = (...args: string[]) => console.info(`[${new Date().toISOString()}]`, ...args);
+
+const command = program
+  .createCommand("run-pipeline")
+  .summary("Run a TriplyDB pipeline")
+  .usage("[options] <json config file>")
+  .description(
+    "Run a TriplyDB pipeline. TriplyDB pipelines currently supports executing speedy construct queries. A pipeline can run for an arbitrary duration, and for arbitrarey result sizes.",
+  )
+  .addHelpText(
+    "after",
+    `
+    Sample json config:
+    {
+      "queries": [{
+          "name": "accountName/queryName", (Required)
+          "priority": 1, (Optional)
+
+      },{
+          "name": "accountName/queryName",
+          "version": 2 (Optional)
+      }],
+      "sourceDataset": "accountName/datasetName", (Required)
+      "targetDataset": "accountName/datasetName", (Required)
+      "targetGraphName": "graph:default", (Optional)
+      "version": 0.1 (Required)
+    }
+    `,
+  )
+  .option("-t, --token <token>", "TriplyDB access token (default: $TRIPLYDB_TOKEN)")
+  .option(
+    "-a, --account <account>",
+    "Account where query job is created and stored. (default: the account that owns the token, or $TRIPLYDB_ACCOUNT)",
+    defaultTriplyDBAccount,
+  )
+  .option("-u, --url <url>", "Optional: Url of the triply API. (default: the API where the token was created)", String)
+  .option(
+    "--http-proxy <proxy>",
+    "Use HTTP proxy for all requests (default: $HTTP_PROXY)",
+    defaultHttpProxy || undefined,
+  )
+  .option(
+    "--https-proxy <proxy>",
+    "Use HTTP proxy for all requests (default: $HTTPS_PROXY)",
+    defaultHttpsProxy || undefined,
+  )
+
+  .action(async () => {
+    function sanityCheckError(msg: string) {
+      console.error(colors.red(msg));
+      command.outputHelp();
+      process.exit(1);
+    }
+    const options = command.opts<{
+      token: string;
+      account?: string;
+      url?: string;
+      httpProxy?: string;
+      httpsProxy?: string;
+    }>();
+    const [configFile] = command.args;
+    if (!configFile || !configFile.length) sanityCheckError("Missing query job config file");
+    const token = options.token ?? defaultTriplyDBToken;
```

Comment:
Why don't you add this as a default in the `option` call, just like for the account?

Reply:
I didn't (this is not code I wrote). But, if I guess it's because the default is printed in the help text. I'm happy with that, but I recall somebody in our team objecting to that ;)